relay-server/src/endpoints/nel.rs

```diff
@@ -0,0 +1,69 @@
+//! Endpoint for Network Error Logging (NEL) reports.
+//!
+//! It split list of incoming events from the envelope into separate envelope with 1 item inside.
+//! Which later get failed by the service infrastructure.
+
+use axum::extract::{DefaultBodyLimit, FromRequest};
+use axum::http::StatusCode;
+use axum::response::IntoResponse;
+use axum::routing::{post, MethodRouter};
+use bytes::Bytes;
+use relay_config::Config;
+use relay_event_schema::protocol::EventId;
+use serde_json::value::RawValue;
+
+use crate::endpoints::common::{self, BadStoreRequest};
+use crate::envelope::{ContentType, Envelope, Item, ItemType};
+use crate::extractors::{Mime, RequestMeta};
+use crate::service::ServiceState;
+
+#[derive(Debug, FromRequest)]
+#[from_request(state(ServiceState))]
+struct NelReportParams {
+    meta: RequestMeta,
+    body: Bytes,
+}
+
+fn is_nel_mime(mime: Mime) -> bool {
+    let ty = mime.type_().as_str();
+    let subty = mime.subtype().as_str();
+    let suffix = mime.suffix().map(|suffix| suffix.as_str());
+
+    matches!(
+        (ty, subty, suffix),
+        ("application", "json", None) | ("application", "reports", Some("json"))
+    )
+}
+
+/// Handles all messages coming on the NEL endpoint.
+async fn handle(
+    state: ServiceState,
+    mime: Mime,
+    params: NelReportParams,
+) -> Result<impl IntoResponse, BadStoreRequest> {
+    if !is_nel_mime(mime) {
+        return Ok(StatusCode::UNSUPPORTED_MEDIA_TYPE.into_response());
+    }
+
+    let items: Vec<&RawValue> =
```

Comment:
In practice, allocating a vector here should be fine. However, since there's currently no limit placed on the number of NEL reports in an envelope - neither in the endpoint nor where it gets sent into validation (`common.rs`) -- let's please be sure we have enough visibility into this endpoint to add more safety measures in the future.

If this turns out to be a problem, we can easily make this a streaming deserializer.